### ISSUE TITLE
Inconsistent Descriptor Space Parsing

### ISSUE THREAD

**Issue by Pttn** [bitcoin/bitcoin#28943]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current behaviour

Sometimes, it is possible to have trailing spaces in a Descriptor and sometimes not. For example:

* `getdescriptorinfo "wsh(multi(2, KzLnMBHHwTs7uUo9UfUxCboj5Urvigx2v7t6iAqhBk2PM4fBMcig , L1S2VZsNYexzyH76CPi1EvrNSoEHsiT12UaR4BTiycmCKqRnSqLY ))"`
Output:
```
{
  "descriptor": "wsh(multi(2,03147a8994e451cb10891688c9db15aea6d62a2691c781678bf7fb3a36e58ccca6,0366b8ab537286e57187eaa8eff803621271500ec496e1e03ef40b9cca6c7a1d1c))#x3mm5nm9",
  "checksum": "xxgjqes0",
  "isrange": false,
  "issolvable": true,
  "hasprivatekeys": true
}
```

* `getdescriptorinfo "wsh( multi(2, KzLnMBHHwTs7uUo9UfUxCboj5Urvigx2v7t6iAqhBk2PM4fBMcig , L1S2VZsNYexzyH76CPi1EvrNSoEHsiT12UaR4BTiycmCKqRnSqLY ) )"`
Output:
```
A function is needed within P2WSH (code -5)
```

* `getdescriptorinfo "wsh(multi(2, 03147a8994e451cb10891688c9db15aea6d62a2691c781678bf7fb3a36e58ccca6, 0366b8ab537286e57187eaa8eff803621271500ec496e1e03ef40b9cca6c7a1d1c))"`
Output:
```
Multi: key ' 03147a8994e451cb10891688c9db15aea6d62a2691c781678bf7fb3a36e58ccca6' is not valid (code -5)
```

### Expected behaviour

Consistence: either never allow spaces, or always allow them, but not both.

### Steps to reproduce

Open Bitcoin-Qt Console and run the commands above.

### How did you obtain Bitcoin Core

Compiled from source

### What version of Bitcoin Core are you using?

master@5f9fd11680af43b0d0d80a6aa4f315aa04afeac4

### Operating system and version

Debian 12

**Comment by fanquake** [comment]
cc @achow101 @Sjors

**Comment by Sjors** [comment]
I think we should disallow them, though BIP 380 doesn't say anything about it.

**Comment by achow101** [comment]
Agree that we should probably just disallow spaces. @sipa any thoughts?

**Comment by achow101** [comment]
> Is it possible someone has a descriptor wallet which has imported descriptors that include spaces?

I don't think that will matter since we rebuild the string when writing to the wallet rather than using what was provided. This should eliminate any spaces.

**Comment by Pttn** [comment]
Fixed by #31603 I think.
